examples/maxmod/streaming/source/main.c

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: CC0-1.0
+//
+// SPDX-FileContributor: Michele Di Giorgio, 2024
+//
+// Example usage of maxmod streaming. We synthesize a sine wave and play it
+// through the maxmod streaming utilities.
+
+#include <stdio.h>
+
+#include <maxmod9.h>
+#include <nds.h>
+
+const int waveFreq = 440;
+int       x        = 0; // wave position
+
+mm_word sineWaveCallback(mm_word length,
+                         mm_addr dest,
+                         mm_stream_formats format)
+{
+    s16 *output = (s16*)dest;
+    for (int i = 0; i < length; ++i) {
+        // Compute value of sine wave at current position
+        const int sine = sinLerp(x);
+
+        *output = sine;
+        output++;
+        *output = -sine;
+        output++;
+
+        x += waveFreq;
+    }
+    return length;
+}
+
+int main(int argc, char **argv)
+{
+    consoleDemoInit();
+
+    printf("maxmod streaming example\n");
+    printf("===========================\n");
+    printf("\n");
+    printf("START: Return to loader\n");
+
+    // Soundbank won't be provided so we need to initialize maxmod without it.
+    mm_ds_system mmSys;
+    mmSys.mod_count    = 0;
+    mmSys.samp_count   = 0;
+    mmSys.mem_bank     = 0;
+    mmSys.fifo_channel = FIFO_MAXMOD;
+    mmInit(&mmSys);
+
+    // Open the stream
+    mm_stream stream;
+    stream.sampling_rate = 25000;
```

Comment:
Where d these values originate from?

Reply:
I've changed the example to play a WAV track which should be more useful